### ISSUE TITLE
json url has changed, and json format has changed

### ISSUE THREAD

**Issue by dimitrihilverda** [igorsuhinin/modx-instagram-latest-posts#6]
The url to the json does not work any more, the new url should be:

// Set the JSON URL
        $jsonUrl = $this->accountUrl . '/?__a=1';

But the returned json is also different , the code needs to be changed to reflect the new locations.

**Comment by igorsuhinin** [comment]
Thanks, @dimitrihilverda. Will check the changes and update the code as needed.

**Comment by dimitrihilverda** [comment]
the video url needs to be retrived with use of the code
$itemsArray = $feed['user']['media']['nodes'];
foreach ($itemsArray as $key => $value) {
$jsonVideo = @file_get_contents('https://www.instagram.com/p/'.$value['code'].'/?__a=1');
} 

(psuedo code)

**Comment by Podryvnik** [comment]
К сожалению, проблема до сих пор актуальна :-( 
Никаких пока мыслей не появилось?

**Comment by igorsuhinin** [comment]
Hi guys,

Just wanted to let you know that currently I'm working on the update. I believe the code can be fixed to follow the recent changes on Instagram so the application will work the same like before.

**Comment by GiveMeGimmicks** [comment]
Do you have an approximate timescale to fix? Client complaining to me their feed has not been displaying for days since json change. I have said it is due to Instagram updates but it would be good to have a timescale... :)

**Comment by igorsuhinin** [comment]
@GMGCreative I hope to have it today.

Looks like after the last changes on Instagram the coming version of this snippet will be able to show 12 last posts only and no videos...

There IS some workaround which should allow to take more posts (haven't tested it fully but I could get 200+ posts without any problem) but it will take more time to develop and it seems it's against Instagram ToS... So I'm not sure if I need to release it at all.

**Comment by GMGCreative** [comment]
@igorsuhinin I think I read that the limit was 12 when first trying to implement a feed. If Instagram state a limit, then that's fine as no one can argue with it :)

**Comment by igorsuhinin** [comment]
Hi guys,

I've updated the snippet. Most likely your websites should work fine with the updated code. I tried to leave all parameters the same as much as possible. Two parameters were removed at all: imageQuality and videoQuality as there is just one file can be returned from Instagram now. I don't think it should create any issues in your code. Anyway I will appreciate any feedback.

**Comment by igorsuhinin** [comment]
Can anyone confirm that everything is good now? I'm going to close the issue so just wanted to be sure.

**Comment by GMGCreative** [comment]
@igorsuhinin Thanks - i've updated the snippet and all seems to be working for me now...

**Comment by igorsuhinin** [comment]
Okay, great. Closing issue now.

**Comment by Podryvnik** [comment]
Огромное спасибо!!! Заработало!!!

**Comment by lottaar** [comment]
The feed is still not working for me. Not getting the error message anymore but no images are shown. All I've done is change the code in the snippet "InstagramLatestPosts" to version 1.5.0. Do I need to change something else as well?

**Comment by igorsuhinin** [comment]
@lottaar You don't need to hide the videos (well, at least if you don't want to use them) with &showVideo=0 parameter. I believe your problem was in the cached data. You probably need to remove the directory /core/cache/instagram_latest_posts/. I do not recommend to not use the cache as Instagram can probably ban your website/IP address if you make many requests to their servers without caching.

**Comment by lottaar** [comment]
I've now deleted the cache file and you're right, now it works with the cache enabled.

However, on another site where there are videos in the feed, no images are shown in the feed when I set showVideo to 1. For some reason it tries to load videos for all of the post, even the image posts. In the source code I get this even when it should be an image: <video><source src="" type="video/mp4"></video>. My innerTpl looks like this:

        [[+type:is=`image`:then=`
        <img src="[[+url]]" alt="" class="img-responsive">
        `]]
        [[+type:is=`video`:then=`
        <video><source src="[[+url]]" type="video/mp4"></video>
        `]]
